mantidproject/mantid
# 21248 renderer cleanup

**NB. This will be merged into a long running feature branch #21230**
This work involves a code cleanup around Geometry Renderers/Handlers which will allow for extending capabilities to perform volume rendering (required for voxel detectors in ESS instruments). This also helps to clarify the interplay between objects (and components) and the associated handler/renderer. Although many instances of unnecessary virtualisation were removed, I do not expect major speed improvements from this PR.

Points to note:
- The `Renderer` class centralizes rendering of all geometry types (replaces `OCGeometryRenderer` etc)
- The `GeometryTriangulator` class handles triangulation of non-standard types (replaces `OCGeometryGenerator` and `CacheGeometryGenerator`)
- The `ShapeInfo` class stores object shape information (replaces `GluGeometryHandler` specifics)
- The `GeometryHandler` class has been modified, is no longer a base class, and handles all geometry types, hence the removal of virtualization (consolidates `CacheGeometryHandler`, `BitmapGeometryHandler`, `StructuredGeometryHandler`, `GluGeometryHandler`, `OCGeometryHandler`)

**To test:**

-Thorough code review.
- Ensure all unit tests related to Object pass
- Ensure instruments are still displayed correctly in the instrument view.

Fixes #21248 . 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
